http://cve.mitre.org/cgi-bin/cvename.cgi?name=CAN-2004-0976
    Multiple scripts in the perl package in Trustix Secure Linux 1.5
    through 2.1, and possibly other operating systems, allows local
    users to overwrite files via a symlink attack on temporary files.

--- lib/CGI/Cookie.pm	2001-03-04 06:53:20.000000000 +1100
+++ lib/CGI/Cookie.pm	2004-12-22 22:29:46.000000000 +1100
@@ -363,7 +363,7 @@
 You may also retrieve cookies that were stored in some external
 form using the parse() class method:
 
-       $COOKIES = `cat /usr/tmp/Cookie_stash`;
+       $COOKIES = `cat /var/run/www/Cookie_stash`;
        %cookies = parse CGI::Cookie($COOKIES);
 
 =head2 Manipulating Cookies

--- lib/ExtUtils/MakeMaker.pm	2001-02-23 13:57:55.000000000 +1100
+++ lib/ExtUtils/MakeMaker.pm	2004-12-22 22:33:57.000000000 +1100
@@ -898,7 +898,7 @@
 The Makefile to be produced may be altered by adding arguments of the
 form C<KEY=VALUE>. E.g.
 
-  perl Makefile.PL PREFIX=/tmp/myperl5
+  perl Makefile.PL PREFIX=~/myperl5
 
 Other interesting targets in the generated Makefile are
 

--- lib/Shell.pm.orig	2004-06-01 11:42:17 +0200
+++ lib/Shell.pm	2005-01-11 15:08:51 +0100
@@ -127,7 +127,7 @@
    use Shell qw(cat ps cp);
    $passwd = cat('</etc/passwd');
    @pslines = ps('-ww'),
-   cp("/etc/passwd", "/tmp/passwd");
+   cp("/etc/passwd", "/etc/passwd.orig");
 
    # object oriented 
    my $sh = Shell->new;

--- lib/perl5db.pl.orig	2004-11-17 05:51:18.000000000 -0700
+++ lib/perl5db.pl	2005-02-08 10:26:10.000000000 -0700
@@ -215,7 +215,7 @@
 =item * noTTY 
 
 if set, goes in NonStop mode.  On interrupt, if TTY is not set,
-uses the value of noTTY or F</tmp/perldbtty$$> to find TTY using
+uses the value of noTTY or F<$HOME/.perldbtty$$> to find TTY using
 Term::Rendezvous.  Current variant is to have the name of TTY in this
 file.
 
@@ -6004,8 +6004,8 @@
             eval "require Term::Rendezvous;" or die;
 
             # See if we have anything to pass to Term::Rendezvous.
-            # Use /tmp/perldbtty$$ if not.
-            my $rv = $ENV{PERLDB_NOTTY} || "/tmp/perldbtty$$";
+            # Use $HOME/.perldbtty$$ if not.
+            my $rv = $ENV{PERLDB_NOTTY} || "$ENV{HOME}/.perldbtty$$";
 
             # Rendezvous and get the filehandles.
             my $term_rv = new Term::Rendezvous $rv;
